bitchat/BitchatApp.swift

```diff
@@ -84,6 +84,9 @@ struct BitchatApp: App {
                         }
                         // Proactively disconnect Nostr to avoid spurious socket errors while Tor is down
                         NostrRelayManager.shared.disconnect()
+                        // Force save identity state (verifications, favorites, etc) before backgrounding
+                        // iOS may terminate the app from background without calling applicationWillTerminate
+                        chatViewModel.saveIdentityState()
```

Comment:
instead of waiting for the scenePhase change we could listen to the "will terminate" notification directly - https://stackoverflow.com/a/67917232

Or, given that we already have an AppDelegate, we could utilize that as well for any before-termination logic (instead of depending on the notification system)